src/beacon.js

```diff
@@ -90,8 +94,8 @@ export default function Emitter(endpoint, context, options) {
         let editedMessage = message;
         editedMessage = message.payload || {};
         editedMessage.type = message.type;
-
-        if (!send(endpoint, JSON.stringify(editedMessage), sync)) {
+        if (!send(endpoint, JSON.stringify(editedMessage), sync) && retries) {
+          --retries
```

Comment:
Nitpick and not wrong, but for clarity we ought to prefer `retries -= 1`.
```suggestion
          retries -= 1;
```